File: templates/backend-typescript-express/src/routes/items.ts
import { Router, Request, Response } from 'express';
<% if (database === 'mongodb') { %>import mongoose from 'mongoose';
import { Item } from '../models/item';
<% } %>
import * as ItemStore from '../models/item';

const router = Router();

<% if (database === 'mongodb') { %>
router.get('/', async (req: Request, res: Response) => {
  try {
    if (mongoose.connection.readyState === 1) {
      const items = await Item.find().sort({ createdAt: -1 });
      return res.json(items);
    }
  } catch (err) {}
  res.json(ItemStore.getAll());
});

router.get('/:id', async (req: Request, res: Response) => {
  try {
    if (mongoose.connection.readyState === 1) {
      const item = await Item.findById(req.params.id);
      if (!item) {
        res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
        return;
      }
      return res.json(item);
    }
  } catch (err) {}
  const item = ItemStore.getById(req.params.id);
  if (!item) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.json(item);
});

router.post('/', async (req: Request, res: Response) => {
  if (!req.body.title) {
    res.status(400).json({ error: { message: 'Title is required', statusCode: 400 } });
    return;
  }
  try {
    if (mongoose.connection.readyState === 1) {
      const item = await Item.create({
        title: req.body.title,
        description: req.body.description || '',
      });
      return res.status(201).json(item);
    }
  } catch (err) {}
  const item = ItemStore.create({
    title: req.body.title,
    description: req.body.description,
  });
  res.status(201).json(item);
});

router.put('/:id', async (req: Request, res: Response) => {
  try {
    if (mongoose.connection.readyState === 1) {
      const item = await Item.findByIdAndUpdate(req.params.id, req.body, { new: true });
      if (!item) {
        res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
        return;
      }
      return res.json(item);
    }
  } catch (err) {}
  const item = ItemStore.update(req.params.id, req.body);
  if (!item) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.json(item);
});

router.delete('/:id', async (req: Request, res: Response) => {
  try {
    if (mongoose.connection.readyState === 1) {
      const item = await Item.findByIdAndDelete(req.params.id);
      if (!item) {
        res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
        return;
      }
      return res.status(204).send();
    }
  } catch (err) {}
  const removed = ItemStore.remove(req.params.id);
  if (!removed) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.status(204).send();
});
<% } else { %>
router.get('/', (req: Request, res: Response) => {
  res.json(ItemStore.getAll());
});

router.get('/:id', (req: Request, res: Response) => {
  const item = ItemStore.getById(req.params.id);
  if (!item) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.json(item);
});

router.post('/', (req: Request, res: Response) => {
  if (!req.body.title) {
    res.status(400).json({ error: { message: 'Title is required', statusCode: 400 } });
    return;
  }
  const item = ItemStore.create({
    title: req.body.title,
    description: req.body.description,
  });
  res.status(201).json(item);
});

router.put('/:id', (req: Request, res: Response) => {
  const item = ItemStore.update(req.params.id, req.body);
  if (!item) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.json(item);
});

router.delete('/:id', (req: Request, res: Response) => {
  const removed = ItemStore.remove(req.params.id);
  if (!removed) {
    res.status(404).json({ error: { message: 'Item not found', statusCode: 404 } });
    return;
  }
  res.status(204).send();
});
<% } %>

export default router;

File: templates/backend-python-sanic/routes/items.py
from sanic import Blueprint, json
from models.item import Item

items_bp = Blueprint("items", url_prefix="/api/items")

items_store = {}
current_id = 1


@items_bp.get("/")
async def list_items(request):
    return json({"data": [item.to_dict() for item in items_store.values()]})


@items_bp.post("/")
async def create_item(request):
    global current_id
    data = request.json
    if not data or "title" not in data:
        return json({"error": "Title is required"}, status=400)

    item = Item(
        id=current_id,
        title=data["title"],
        description=data.get("description", ""),
        completed=bool(data.get("completed", False)),
    )
    items_store[current_id] = item
    current_id += 1
    return json({"data": item.to_dict()}, status=201)


@items_bp.get("/<item_id:int>")
async def get_item(request, item_id: int):
    item = items_store.get(item_id)
    if not item:
        return json({"error": "Item not found"}, status=404)
    return json({"data": item.to_dict()})


@items_bp.put("/<item_id:int>")
async def update_item(request, item_id: int):
    item = items_store.get(item_id)
    if not item:
        return json({"error": "Item not found"}, status=404)

    data = request.json
    if not data:
        return json({"error": "Invalid request body"}, status=400)

    if "title" in data:
        item.title = data["title"]
    if "description" in data:
        item.description = data["description"]
    if "completed" in data:
        item.completed = bool(data["completed"])

    return json({"data": item.to_dict()})


@items_bp.delete("/<item_id:int>")
async def delete_item(request, item_id: int):
    if item_id not in items_store:
        return json({"error": "Item not found"}, status=404)
    del items_store[item_id]
    return json({"message": "Item deleted"})
